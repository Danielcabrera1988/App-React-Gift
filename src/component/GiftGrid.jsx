import { useFetchGifs } from "../hooks/useFetchGifs";
import { GiftItem } from "./GiftItem";

export const GiftGrid = ({ category }) => {
  //CREANDO EL PRIMER CUSTOM HOOKS
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && <h2>Cargando...</h2> //usando and l√≥gico
      }
      <div className="card-grid">
        {images.map((image) => (
          <GiftItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
