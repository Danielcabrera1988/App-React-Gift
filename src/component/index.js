//Esta es una manera de simplificar las importaciones, usando un archivo que haga referencia a los demas archivos o componentes que voy a necesitar, se los conoce como archivo barril.
export * from "./AddCategories";
export * from "./GifGrid";
export * from "./GifItem";
