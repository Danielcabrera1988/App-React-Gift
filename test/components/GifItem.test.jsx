import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/component/GifItem";

describe("Comprobando GiftItem", () => {
  const title = "PELONDRA";
  const url = "http://algopelado.com/pelado.jpg";
  test("Debe realziar la snap del componente", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la img con el url y el alt indicado", () => {
    render(<GifItem title={title} url={url} />);

    expect( screen.getByRole('img').src ).toBe(url)
      
  });
});
