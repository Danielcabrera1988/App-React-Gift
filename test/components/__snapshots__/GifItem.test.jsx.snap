// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comprobando GiftItem Debe realziar la snap del componente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="PELONDRA"
      src="http://algopelado.com/pelado.jpg"
    />
    <p>
      PELONDRA
    </p>
  </div>
</div>
`;
